source/magpie/group.d

```diff
@@ -0,0 +1,154 @@
+module magpie.group;
+
+import magpie.dataframe: DataFrame;
+import magpie.index: Index;
+import magpie.helper: dropper, transposed, toArr, vectorize;
+
+import std.meta: staticMap;
+
+/// Struct for groupBy Operation
+struct Group(GrpRowType...)
+{
+    /// Group names
+    string[][] groups;
+
+    /// number of elements before the particular group
+    int[] elementCountTill;
+
+    alias GrpType = staticMap!(toArr, GrpRowType);
+    /// Data of Group
+    GrpType data;
+
+    /// Index for group
+    Index grpIndex;
+
+public:
+    /++
+    int getGroupPosition(string[] grpTitles)
+    Description: Get position of a particular group in the sruct
+    @params: grpTitles - the group you wnant to search for
+    +/
+    int getGroupPosition(string[] grpTitles)
```

Comment:
not that it really matters, but you should probably return a `ptrdiff_t` here instead of `int`.